src/xenia/cpu/backend/x64/x64_backend.cc

```diff
@@ -116,6 +124,103 @@ std::unique_ptr<GuestFunction> X64Backend::CreateGuestFunction(
   return std::make_unique<X64Function>(module, address);
 }
 
+bool X64Backend::InstallBreakpoint(Breakpoint* bp) {
+  auto functions = processor()->FindFunctionsWithAddress(bp->address());
+  if (functions.empty()) {
+    // Go ahead and fail - let the caller handle this.
+    return false;
+  }
+
+  for (auto function : functions) {
+    assert_true(function->is_guest());
+    auto guest_function = reinterpret_cast<cpu::GuestFunction*>(function);
+    auto code = guest_function->MapGuestAddressToMachineCode(bp->address());
+    if (!code) {
+      // This should not happen.
+      assert_always();
+      return false;
+    }
+
+    auto orig_bytes =
+        xe::load_and_swap<uint16_t>(reinterpret_cast<void*>(code + 0x0));
+    bp->backend_data().push_back({code, orig_bytes });
+
+    xe::store_and_swap<uint16_t>(reinterpret_cast<void*>(code + 0x0), 0x0F0C);
+  }
+
+  return true;
+}
+
+bool X64Backend::InstallBreakpoint(Breakpoint* bp, Function* func) {
+  assert_true(func->is_guest());
+  auto guest_function = reinterpret_cast<cpu::GuestFunction*>(func);
+  auto code = guest_function->MapGuestAddressToMachineCode(bp->address());
+  if (!code) {
+    assert_always();
+    return false;
+  }
+
+  // Assume we haven't already installed a breakpoint in this spot.
+  auto orig_bytes =
+      xe::load_and_swap<uint16_t>(reinterpret_cast<void*>(code + 0x0));
+  bp->backend_data().push_back({code, orig_bytes});
+
+  xe::store_and_swap<uint16_t>(reinterpret_cast<void*>(code + 0x0), 0x0F0C);
+  return true;
+}
+
+bool X64Backend::UninstallBreakpoint(Breakpoint* bp) {
+  for (auto pair : bp->backend_data()) {
```

Comment:
(unsure) in general you may want to do `auto&` so that the pair is not copied - I always forget if the compiler implicitly handles that or not, but I do know it'll often copy things